### PR TITLE
Use PackageDownload

### PR DESCRIPTION
**Pull request by dsplaisted**
Fix https://github.com/dotnet/cli/issues/10440 (we will need https://github.com/dotnet/project-system/issues/4567 to be able to support this in VS).

### CODE REVIEW

**Comment by rainersigwald** [comment]
@chcosta Jinx!

File: src/Tasks/Microsoft.NET.Build.Tasks/ResolveAppHosts.cs [hunk 1]
```diff
@@ -19,6 +19,8 @@ public class ResolveAppHosts : TaskBase
 
         public string AppHostRuntimeIdentifier { get; set; }
 
+        public string [] OtherRuntimeIdentifiers { get; set; }
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/ResolveAppHosts.cs]: nit, extra space between `string` and `[]`

**Review by nguerrera — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.TargetingPackResolution.targets [hunk 2]
```diff
@@ -107,22 +107,34 @@ Copyright (c) .NET Foundation. All rights reserved.
...
 
-    <ItemGroup>
-      <PackageReference Include="@(_PackageReferenceToAdd)"
+    <PropertyGroup Condition="'$(UsePackageDownload)' == ''">
+      <UsePackageDownload Condition=" '$(MSBuildRuntimeType)' == 'Core'">true</UsePackageDownload>
+      <UsePackageDownload Condition=" '$(MSBuildRuntimeType)' != 'Core'">false</UsePackageDownload>
+    </PropertyGroup>
+    
+    <ItemGroup Condition="'$(UsePackageDownload)' == 'true'">
+      <PackageDownload Include="@(_PackageToDownload)">
```
Comment (wli3) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.TargetingPackResolution.targets]: just help me understanding this nuget feature, we use PackageDownload to download, but i didn't find the code to dig (i think it is in)nuget cache.

Comment (dsplaisted) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.TargetingPackResolution.targets]: It's the `GetPackageDirectory` task.  It doesn't change with this PR, with `PackageDownload`, the packages are downloaded to the same place that the would have been with `PackageReference`.

**Pull request merged**